### PR TITLE
Self-serve unwithdrawing for managing editors

### PR DESCRIPTION
**Pull request by benilovj**
With this PR, managing editors can unwithdraw editions through the Whitehall Admin. Since #2489, unwithdrawing is a service within Whitehall.

After this change, unwithdrawing using the rake task will no longer be possible.

Considerations for the review:

- is the microcopy appropriate? (see the screenshots below)
- there is no Google Analytics instrumentation yet for the unwithdraw action, but this probably should be added on a separate PR ( @dsingleton are you aware of any helpers in Whitehall to make sending events to GA easier?)

## Button in the admin view

image

## Modal prompt

image

## Full-page prompt

image

## Remarks on the edition

image

[Story in Trello](https://trello.com/c/jkI34gB3/303-build-a-self-service-option-for-un-withdrawing-documents-in-whitehall-for-managing-editors-medium)

### CODE REVIEW

File: features/step_definitions/unpublishing_published_documents_steps.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@
   end
 end
 
-When(/^I withdraw the publication because it is no longer government publication$/) do
+When(/^I withdraw the publication because it is no longer reflects current government policy$/) do
```
Comment (benlovell) [inline features/step_definitions/unpublishing_published_documents_steps.rb]: This doesn't make sense with the 'is' present.

**Comment by boffbowsh** [comment]
Click tracking is provided by the `track_click.js` module in the govuk_admin_template https://github.com/alphagov/govuk_admin_template/blob/master/app/assets/javascripts/govuk-admin-template/modules/track_click.js. See https://github.com/alphagov/signonotron2/blob/e06a627ef7669090274e8a5af14f4ee142928cd0/app/helpers/analytics_helper.rb for an example

File: features/unwithdrawing-withdrawn-documents.feature [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Feature: Unpublishing published documents
+  As a Managing Editor
+  I can unwithdraw unwithdrawn documents
+  So that I can adjust them with banners or annotations and withdraw them again
+
+  Scenario: Unwithdrawing a withdrawn document
+    Given I am a managing editor
+    And a published publication "Shaving kits for all" exists
+    When I withdraw the publication because it is no longer reflects current government policy
```
Comment (benlovell) [inline features/unwithdrawing-withdrawn-documents.feature]: Same here of course.

**Comment by boffbowsh** [comment]
Maybe some help text about what this action actually does would be useful.
"This will create a new edition from the withdrawn edition and immediately publish it"

File: features/step_definitions/unpublishing_published_documents_steps.rb (hunk 1 above)
Comment (benilovj) [inline features/step_definitions/unpublishing_published_documents_steps.rb]: thanks, fixed.

**Comment by benilovj** [comment]
@boffbowsh good idea, I've added the help text and updated the screenshots.

**Comment by boffbowsh** [comment]
:+1: from me, worth getting approval from Ga an

**Comment by benlovell** [comment]
:rocket: :ship:

**Comment by benilovj** [comment]
Merging as this has been okayed by Jen and Gavan...

**Pull request merged**